controllers/organization_controller.go

```diff
@@ -0,0 +1,157 @@
+
+	log.V(4).Info("Getting Organization and Members..")
+	org, orgMemb, err := r.getOrganizationAndMembers(ctx, req.NamespacedName)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !org.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.V(4).Info("Deleting Keycloak group..")
+		err = r.Keycloak.DeleteGroup(ctx, org.Name)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		err = r.removeFinalizer(ctx, org, orgMemb)
+		return ctrl.Result{}, err
+	}
+	err = r.addFinalizer(ctx, org, orgMemb)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	group := buildKeycloakGroup(org, orgMemb)
+
+	log.V(4).Info("Reconciling Keycloak group..")
+	group, err = r.Keycloak.PutGroup(ctx, group)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	log.V(4).Info("Updating status..")
+	err = r.updateOrganizationStatus(ctx, org, orgMemb, group)
+	return ctrl.Result{}, err
+}
+
+func (r *OrganizationReconciler) getOrganizationAndMembers(ctx context.Context, orgKey types.NamespacedName) (*orgv1.Organization, *controlv1.OrganizationMembers, error) {
+	org := &orgv1.Organization{}
+	if err := r.Get(ctx, orgKey, org); err != nil {
+		return org, nil, err
+	}
+
+	memb := &controlv1.OrganizationMembers{}
+	membKey := types.NamespacedName{
+		Namespace: org.Name,
+		Name:      "members",
+	}
+	err := r.Get(ctx, membKey, memb)
+	return org, memb, err
+}
+
+func (r *OrganizationReconciler) addFinalizer(ctx context.Context, org *orgv1.Organization, memb *controlv1.OrganizationMembers) error {
+	if !controllerutil.ContainsFinalizer(memb, orgFinalizer) {
+		controllerutil.AddFinalizer(memb, orgFinalizer)
+		if err := r.Update(ctx, memb); err != nil {
+			return err
+		}
+	}
+	if !controllerutil.ContainsFinalizer(org, orgFinalizer) {
+		controllerutil.AddFinalizer(org, orgFinalizer)
+		if err := r.Update(ctx, org); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (r *OrganizationReconciler) removeFinalizer(ctx context.Context, org *orgv1.Organization, memb *controlv1.OrganizationMembers) error {
+	if controllerutil.ContainsFinalizer(org, orgFinalizer) {
+		controllerutil.RemoveFinalizer(org, orgFinalizer)
+		if err := r.Update(ctx, org); err != nil {
+			return err
+		}
+	}
+	if controllerutil.ContainsFinalizer(memb, orgFinalizer) {
+		controllerutil.RemoveFinalizer(memb, orgFinalizer)
+		if err := r.Update(ctx, memb); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *OrganizationReconciler) updateOrganizationStatus(ctx context.Context, org *orgv1.Organization, memb *controlv1.OrganizationMembers, group keycloak.Group) error {
+	userRefs := []controlv1.UserRef{}
+	for _, u := range group.Members {
+		userRefs = append(userRefs, controlv1.UserRef{
+			Name: u,
+		})
+	}
+	memb.Status.ResolvedUserRefs = userRefs
+	return r.Status().Update(ctx, memb)
+}
+
+func buildKeycloakGroup(org *orgv1.Organization, memb *controlv1.OrganizationMembers) keycloak.Group {
+	groupMem := []string{}
```

Comment:
```suggestion
	groupMem := make([]string, 0, len(memb.Spec.UserRefs))
```

Not that it really matters...